stages/org.osbuild.dracut

```diff
@@ -0,0 +1,147 @@
+    }
+  },
+  "omit_modules": {
+    "description": "Dracut modules to not include.",
+    "type": "array",
+    "items": {
+      "type": "string",
+      "description": "A dracut module, e.g. base, nfs, network ..."
+    }
+  },
+  "drivers": {
+    "description": "Kernel modules to exclusively include.",
+    "type": "array",
+    "items": {
+      "type": "string",
+      "description": "A kernel module without the .ko extension"
+    }
+  },
+  "add_drivers": {
+    "description": "Add a specific kernel modules.",
+    "type": "array",
+    "items": {
+      "type": "string",
+      "description": "A kernel module without the .ko extension"
+    }
+  },
+  "force_drivers": {
+    "description": "Add driver and ensure that they are tried to be loaded.",
+    "type": "array",
+    "items": {
+      "type": "string",
+      "description": "A kernel module without the .ko extension"
+    }
+  },
+  "filesystems": {
+    "description": "Kernel filesystem modules to exclusively include.",
+    "type": "array",
+    "items": {
+      "type": "string",
+      "description": "A kernel module without the .ko extension"
+    }
+  },
+  "install": {
+    "description": "Add custom files to the initramfs.",
+    "type": "array",
+    "items": {
+      "type": "object",
+      "description": "What (keys) to include where (values)"
+    }
+  }
+}
+"""
+
+
+def main(tree, options):
+    modules = options.get("modules", [])  # dracut modules
+    add_modules = options.get("add_modules", [])
+    omit_modules = options.get("omit_modules", [])
+    drivers = options.get("drivers", [])  # kernel modules
+    add_drivers = options.get("add_drivers", [])
+    force_drivers = options.get("force_drivers", [])
+    filesystems = options.get("filesystems", [])
+    install = options.get("install", [])
+
+    # internal option for exploring purposes
+    host_only = options.get("host-only", False)
+
+    # initrds have already been created, force the recreation
+    options = ["--force", "-v", "--show-modules"]
+
+    options += ["--host-only" if host_only else "--no-host-only"]
+
+    if modules:
+        options += ["--modules", " ".join(modules)]
+
+    if add_modules:
+        options += ["--add", " ".join(add_modules)]
+
+    if omit_modules:
+        options += ["--omit", " ".join(omit_modules)]
+
+    if drivers:
+        options += ["--drivers", " ".join(drivers)]
+
+    if add_drivers:
+        options += ["--add-drivers", " ".join(add_drivers)]
+
+    if force_drivers:
+        options += ["--force-drivers", " ".join(force_drivers)]
+
+    if filesystems:
+        options += ["--filesystems", " ".join(filesystems)]
+
+    if install:
+        for l in install:
+            for k, v in l.items():
+                options += ["--install", k, v]
+
+    # discover the installed kernels via their modules
+    # this way we also ignore the installed rescue kernel
+    for dirent in os.scandir(f"{tree}/usr/lib/modules"):
```

Comment:
Do we want to mention these in the options explicitly at some point? Even if we don't do it now, does it make sense to allow passing all the above options separately for each kernel? Or would we always want the same for all installed kernels?

Reply:
The only exception I can think of is Xen, some others use Fedora with it (eg. the QubeOS project) so it's possible that different types of kernels would need a different set of options.

Reply:
Would not be hard to put everything in a another level of dictionary for `"kernel-version-xzy": { ... options ... }` and then maybe now have a wildcard option `"*": { ... options ... }`. Complicates things a bit and can lead to two different pipelines having the same output, but we can see it as a transitional step.

Reply:
Another option would be to just keep the top level options (or stick them into a `default` section) and then have per kernel overrides?

Reply:
per-kernel overrides makes sense, I don't think it is very common to use something different so may as well not complicate it more than it needs to be.

Reply:
Hm sorry I don't see the difference between your two options. Having a `default` or `*` catch-al option amounts to the same, no?

Reply:
Sorry for not being clear enough:
```json
"default" {
  "modules": ["base", "networking"]
},
"kernel-abc": {
  "filesystem": ["xfs"]
}
```

There for `kernel-abc`  both `"modules"`  (`["base", "networking"]`) and `"filesystem"` (`["xfs"]`) would apply, while for `kernel-xyz` only `"modules": ["base", "networking"]`. 
OTOH:
```json
"*" {
  "modules": ["base", "networking"]
},
"kernel-abc": {
  "filesystem": ["xfs"]
}
```
There `kernel-abc` would have only `"filesystem"` (`["xfs"]`) set (`kernel-xyz` still has only `"modules": ["base", "networking"]`).

Reply:
ahh! I prefer the `*`, because it seems more explicit and harder to interpret wrongly to me. Sorry for kicking off a discussion about something that we don't even know we need yet...

Reply:
In the new version we are actually passing the kernel versions in, and I figured if one needs different options for different kernels, just add the stage twice. Sounds ok?